r/R/dplyr.R

```diff
@@ -619,9 +619,21 @@ collect.arrow_dplyr_query <- function(x, as_data_frame = TRUE, ...) {
     restore_dplyr_features(tab, x)
   }
 }
-collect.ArrowTabular <- as.data.frame.ArrowTabular
+collect.ArrowTabular <- function(x, as_data_frame = TRUE, ...) {
+  if (as_data_frame) {
+    as.data.frame(x, ...)
+  } else {
+    x
+  }
+}
 collect.Dataset <- function(x, ...) dplyr::collect(arrow_dplyr_query(x), ...)
 
+compute.arrow_dplyr_query <- function(x, ...) dplyr::collect(x, as_data_frame = FALSE)
+compute.ArrowTabular <- function(x, ...) x
+compute.Dataset <- function(x, ...) {
+  dplyr::collect(arrow_dplyr_query(x), as_data_frame = FALSE, ...)
+}
```

Comment:
```suggestion
compute.Dataset <- compute.arrow_dplyr_query
```

Reply:
But don't we need to convert the `Dataset` to an `arrow_dplyr_query` before passing it to `compute()`?

Reply:
I could do this instead:
```suggestion
compute.Dataset <- function(x, ...) dplyr::compute(arrow_dplyr_query(x), ...)
```

Reply:
You don't need that because `collect.Dataset` wraps it for you

Reply:
Ohhh, right, thanks. Fixed in 95ecb18ab87421369f90ae309d423dd84c7ee240